README.md

```diff
@@ -14,30 +14,183 @@ jobs:
 
     strategy:
       matrix:
-        tutor_version: ["<17.0.0", "==17.0.3", "<18.0.0"]
+        tutor_version: ["<17.0.0", "==17.0.3", "<18.0.0", "nightly"]
 
     steps:
-      - uses: actions/checkout@v4
-        with:
-          path: eox-hooks
-
-      - uses: eduNEXT/integration-test-in-tutor
+      - name: Run Integration Tests
+        uses: eduNEXT/integration-test-in-tutor@main
         with:
+          app_name: "eox-test"
           tutor_version: ${{ matrix.tutor_version }}
-          app_name: "eox-hooks"
-          shell_file_to_run: "eox_hooks/tests/tutor/integration.sh"
+          shell_file_to_run: "tests/integration.sh"
+          openedx_extra_pip_requirements: "package1==1.0 package2>=2.0"
+          fixtures_file: "fixtures/test_data.json"
+          openedx_imports_test_file_path: "tests/integration/import_tests.py"
+          openedx_imports_test_function_name: "test_openedx_imports"
 ```
 
 ## Inputs
 
 ### `app_name`
 
-**Required** Application name to test. E.g., eox-tenant.
+**Required**  
+The name of your plugin/application to test. This should match the directory name of your plugin.  
+*Example*: `"eox-tenant"`
 
 ### `tutor_version`
 
-**Required** The tutor version matrix to use.
+**Required**  
+The version of Tutor to use. You can specify:
+
+- A specific version number (e.g., `"==17.0.3"`).
+- A comparison operator with a version (e.g., `"<18.0.0"`).
+- The string `"nightly"` to use the latest development version.
+
+*Examples*:
+
+- `"==17.0.3"`
+- `"<18.0.0"`
+- `"nightly"`
 
 ### `shell_file_to_run`
 
-(Optional) The path of the shell file to run the integration tests.
+**Required**  
+The path to the shell script that runs your integration tests. This path is relative to your plugin directory.  
+*Example*: `"tests/integration.sh"`
+
+### `openedx_extra_pip_requirements`
+
+**Optional**  
+Extra pip requirements to install in Open edX. These are additional Python packages your plugin depends on. Provide them as a space-separated string.  
+*Default*: `""` (empty string)  
+*Example*: `"package1==1.0 package2>=2.0"`
+
+### `fixtures_file`
+
+**Optional**  
+The path to the fixtures file of your plugin to load initial data for the tests. This path is relative to your plugin directory.  
+*Example*: `"fixtures/test_data.json"`
+
+### `openedx_imports_test_file_path`
+
+**Optional**  
+The path to the Python file in your plugin that contains the test function for validating Open edX imports. This path is relative to your plugin directory.  
+*Example*: `"tests/import_tests.py"`
+
+### `openedx_imports_test_function_name`
+
+**Optional**  
+The name of the function in the specified file that executes the import tests for Open edX.  
+*Example*: `"test_openedx_imports"`
```

Comment:
@mariajgrimaldi, @BryanttV: 

We previously used pytest to run these tests when the action was executed inside the LMS container.
Initially, I considered using pytest within the LMS container for this test as well, but this approach would require setting up a new virtual environment inside the container and installing test dependencies like pytest and pytest-django. After checking the import test and realizing is a simple function that makes the imports, I thought it might unnecessary setting up pytest when we could just directly invoke the function within the container

I'm interested in hearing your thoughts on this approach

Reply:
Thanks for the additional context, now I understand the implementation better. My only concern with this approach is that maintaining it would mean maintaining another way of running tests besides the Pytest approach. Does installing Pytest add so much overhead?

Reply:
Thanks @magajh. I agree with Majo's question. Also, do you happen to have any log for the jobs that shows what it looks like when some backend fails?

Reply:
@mariajgrimaldi @BryanttV: I agree that running tests in two different ways might not be ideal, and we should stick to only executing the tests with pytest if possible

Here’s an update on the progress I’ve made while setting up pytest:

After testing further, I found that installing `pytest` and `pytest-django` dependencies directly in the container works perfectly (as I did in the [latest commit](https://github.com/eduNEXT/integration-test-in-tutor/pull/4/commits/935eafedd0f2529a80e61e8087606af6793e5885)). However, I don’t think this is the best approach since we are permanently altering the container and affecting the environment where the platform runs—something we want to avoid.

I’ve been trying to create a separate virtual environment to install these dependencies, but that’s where I’ve encountered issues. The virtual environment created for testing does not inherit the globally installed packages in the LMS container, such as Django. I believe this is because in the Docker container, the packages might be installed in paths that aren’t accessible or visible to the new virtual environment. 

I tried using the `--system-site-packages` option when creating the venv to make it recognize global packages, but it didn’t work. So, I ended up installing all the dev dependencies using the target in the Makefile (`install-dev-dependencies`), which worked fine. The problem I still can’t solve is that running pytest in that venv doesn’t work. The key issue seems to be that the virtual environment inside the LMS container is not properly set up to recognize the LMS modules and paths needed for the tests.

I’ve been trying to adjust the `PYTHONPATH` to include the necessary directories when running the tests, but so far, it hasn’t worked.

Here’s an example of the error I’m getting: https://github.com/eduNEXT/eox-core/actions/runs/11150314446/job/30991166191#step:2:7014

I’ll try a few more things to get pytest running in an isolated venv and will update you here.

In case you want to check the branch with the workflow where I’m trying to make this work in the venv, it’s `mjh/test-changes-run-tests-outside-container`

Reply:
Thank you for all the context, @magajh! Have you tried using [`tutor local run`](https://docs.tutor.edly.io/reference/cli/local.html#tutor-local-run)? This command creates a new container, and we would not be affecting the one already running, but I'm not sure it works.

Reply:
> The problem I still can’t solve is that running pytest in that venv doesn’t work. The key issue seems to be that the virtual environment inside the LMS container is not properly set up to recognize the LMS modules and paths needed for the tests.

Did you try running `pip install -e .` inside the venv?

Also, pytest or pytest-django don't share dependencies with what's already installed in the venv for a production environment (base.txt). From what I gathered from the repositories, pytest-django only [requires pytest](https://github.com/pytest-dev/pytest-django/blob/1ffc3239387bcfadbd8aecfdfacf3d6b52b947f8/pyproject.toml#L40-L42), and [pytest requires a list of libraries](https://github.com/pytest-dev/pytest/blob/7add2af7087dc32af19a38f5950a5b64a19d1de5/pyproject.toml#L49-L56) without any specific versions -- at least that conflict with our production environment, some are installed for testing or development, but not in base.txt.

Reply:
@BryanttV since `tutor local run` creates a new container from the image, we would need to install the plugin, the extra pip requirements, run migrations, execute the test for the backends - all in the same run command -, and then do it again with `tutor local exec` to run the integration tests

Reply:
I agree with leaving it as is with pytest. Thank you for addressing all of our concerns!

Reply:
Updated the readme and workflow to remove any trace of the `openedx_imports_test_function_name` input